### PR TITLE
tests: create remote fixtures

### PR DESCRIPTION
**Pull request by efiop**
First steps in migrating to convenient cloud fixtures. External outs/deps tests and `put_object` tests are not converted for now, this PR is a pre-requisite. A lot of the stuff was just adapted for the new approach, remote helpers will be reworked some more in the next PR.

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: tests/func/test_api.py [hunk 1]
```diff
@@ -5,93 +5,42 @@
...
+
+cloud_names = [
+    "s3",
+    "gs",
+    "azure",
+    "gdrive",
+    "oss",
+    "ssh",
+    "hdfs",
+    "http",
```
Comment (efiop) [inline tests/func/test_api.py:19]: Also added http api test.

File: tests/func/test_api.py [hunk 2]
```diff
@@ -5,93 +5,42 @@
...
-        GDrive.create_dir(dvc, url)
-        with dvc.config.edit() as conf:
-            conf["remote"][TEST_REMOTE].update(
-                gdrive_service_account_email="test",
-                gdrive_service_account_p12_file_path="test.p12",
-                gdrive_use_service_account=True,
-            )
-        dvc.scm.add(dvc.config.files["repo"])
-        dvc.scm.commit(f"modify '{TEST_REMOTE}' remote")
-
```
Comment (efiop) [inline tests/func/test_api.py:75]: Finally this moved into the fixture itself. But the fixture itself could be  and will be improved further later.

**Comment by skshetry** [comment]
@efiop, this looks good than before, though I don't like `TmpDir` getting more and more powerful. Feels like we are going the same old route (remember old `erepo_dir`?). 

What I'm saying is, I don't see any difference between:
```py
setup_remote(dvc, config=s3_config)
```

vs:

```python
tmp_dir.add_remote(config=s3_config)
```

**Comment by efiop** [comment]
@skshetry Great point! What I really wanted is something like `dvc.remote.add`, but didn't want to expand the scope of this even further. It just feels like an overkill to have a separate fixture for that, especially since we no longer use it everywhere thanks to `local_remote` fixture, which is more intuitive. TmpDir will likely become a base class for working with external workspaces, just because it is is so convenient, and will get some additional properties/methods, which is not a bad thing as long as it is handy for writing the tests. I'll give it another look in the upcoming externa outs/deps PR.

**Pull request merged**